ui/page/components/consensus_list.go

```diff
@@ -0,0 +1,145 @@
+package components
+
+import (
+	"image/color"
+	"strings"
+
+	"gioui.org/layout"
+	"gioui.org/text"
+	"gioui.org/unit"
+
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+var canVote bool
+
+type ConsensusItem struct {
+	Agenda     dcrlibwallet.Agenda
+	VoteButton decredmaterial.Button
+}
+
+func AgendasList(gtx C, l *load.Load, consensusItem *ConsensusItem) D {
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	agenda := consensusItem.Agenda
+	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return layoutAgendaStatus(gtx, l, consensusItem.Agenda)
+		}),
+		layout.Rigid(layoutAgendaDetails(l, agenda.Description)),
+		layout.Rigid(layoutAgendaDetails(l, "ID: #"+agenda.AgendaID)),
+		layout.Rigid(layoutAgendaDetails(l, "Voting Preference: "+agenda.VotingPreference)),
+		layout.Rigid(func(gtx C) D {
+			return layoutAgendaVoteAction(gtx, l, consensusItem)
+		}),
+	)
+}
+
+func layoutAgendaStatus(gtx C, l *load.Load, agenda dcrlibwallet.Agenda) D {
+
+	var statusLabel decredmaterial.Label
+	var statusIcon *decredmaterial.Icon
+	var backgroundColor color.NRGBA
+
+	switch agenda.Status() {
+	case "Finished":
```

Comment:
Define these as constants in dcrlibwallet and use here instead of string literals. `AgendaStatusFinished = "Finished"` for example.